### ISSUE TITLE
Regex 1.8.0: Valid `?P<foo>` -> `?<foo>` triggers lint

### ISSUE THREAD

**Issue by torsteingrindvik** [rust-lang/rust-clippy#10680]
### Summary

https://github.com/rust-lang/regex/blob/master/CHANGELOG.md
https://github.com/rust-lang/regex/issues/955

A new feature was added to regex 1.8.0 which allows removing this `P` character.
The feature works, but clippy emits a false positive to it.

### Lint Name

clippy::invalid_regex

### Reproducer

```rust
use regex::Regex;

fn main() {
    let _re1 = Regex::new(r"(?P<hi>hi)").unwrap();

    // regex 1.8.0: Can now remove the P character
    let _re2 = Regex::new(r"(?<hi>hi)").unwrap();
    // ^-- clippy::invalid_regex
}

```

I saw this happen: Clippy complains.
I expected to see this happen: No complaints.

### Version

```text
rustc 1.71.0-nightly (8bdcc62cb 2023-04-20)
binary: rustc
commit-hash: 8bdcc62cb0362869f0e7b43a6ae4f96b953d3cbc
commit-date: 2023-04-20
host: x86_64-unknown-linux-gnu
release: 1.71.0-nightly
LLVM version: 16.0.2
```
